RUFAS/routines/field/field/tillage_application.py

```diff
@@ -115,30 +115,42 @@ def till_soil(
             tillage_depth = self.soil_data.soil_layers[-1].bottom_depth
 
         total_phosphorus_incorporated = 0
-        phosphorus_pools_to_draw_from = [
+        non_manure_phosphorus_pools = [
             "available_phosphorus_pool",
             "recalcitrant_phosphorus_pool",
-            "machine_water_extractable_inorganic_phosphorus",
-            "machine_water_extractable_organic_phosphorus",
-            "machine_stable_inorganic_phosphorus",
-            "machine_stable_organic_phosphorus",
-            "grazing_water_extractable_inorganic_phosphorus",
-            "grazing_water_extractable_organic_phosphorus",
-            "grazing_stable_inorganic_phosphorus",
-            "grazing_stable_organic_phosphorus",
         ]
-        for pool in phosphorus_pools_to_draw_from:
+        for pool in non_manure_phosphorus_pools:
             total_phosphorus_incorporated += self._remove_amount_incorporated(
-                self.soil_data, pool, incorporation_fraction
+                self.soil_data, "", pool, incorporation_fraction
             )
+        manure_phosphorus_pools = [
+            "water_extractable_inorganic_phosphorus",
+            "water_extractable_organic_phosphorus",
+            "stable_inorganic_phosphorus",
+            "stable_organic_phosphorus",
+        ]
+        manure_pool_types = [
+            "grazing_manure",
+            "machine_manure",
+        ]
+        for manure_pool_type in manure_pool_types:
+            for pool in manure_phosphorus_pools:
+                total_phosphorus_incorporated += self._remove_amount_incorporated(
+                    self.soil_data, manure_pool_type, pool, incorporation_fraction
+                )
```

Comment:
```suggestion
        manure_pool_types = [
            self.soil_data.grazing_manure,
            self.soil_data.machine_manure,
        ]
        for manure_pool_type in manure_pool_types:
            for pool in manure_phosphorus_pools:
                total_phosphorus_incorporated += self._remove_amount_incorporated(
                    manure_pool_type, pool, incorporation_fraction
                )
```
I think this would make it unnecessary to modify `_remove_amount_incorporated` at all.

Reply:
Ooh, I like this, will give it a try. Nice idea.